presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +259,11 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            LocalDate localDate = DATE_FORMATTER.parseLocalDate(String.valueOf(value));
```

Comment:
Can you please explain in comment why such logic is needed. It may be just me, but it is not obvious why original code does not work. What is the difference.

Reply:
Thanks for pointing that out.  I'll comment the code, but I'll do it here first, so we can discuss.

So the problem was that by default, Joda uses the ISO chronology, which doesn't work for dates before the introduction of the Gregorian calendar.  The same is true for `LocalDateTime` in the new Java API.  Joda allows you to specify a chronology and from the description, `GJChronology` seemed the appropriate one, but it was producing results that were off by an hour.  Perhaps I did something wrong there, but anyway, the old Java calendar worked, so I stuck with it (also, we are moving away from Joda, so relying more on it didn't seem right).

Reply:
By quick google search I do not see a good approach which does not involve Joda (even if that one works, which is not obvious).

Reply:
We are not moving away from Joda-Time everywhere. `java.time` is preferable if it works, but is too expensive to use for many of the time related functions, due to the object creation overhead. (at least, this is how it appears ... it is possible that benchmarks could prove this wrong)